addons/website_sale_loyalty/models/sale_order.py

```diff
@@ -186,3 +186,18 @@ def _gc_abandoned_coupons(self, *args, **kwargs):
         so_to_reset.applied_coupon_ids = False
         for so in so_to_reset:
             so._update_programs_and_rewards()
+
+    @api.model
+    def get_website_sale_extra_values(self, order):
+        """
+        This method provides the additional values for the cart template in the context of the website_sale_loyalty module.
+        :param order: The sale.order object of the current order
+        :return: A dictionary of additional values to be passed to the cart template
+        """
+        promo_code_success = order.get_promo_code_success_message()
+        promo_code_error = order.get_promo_code_error()
```

Comment:
```suggestion
        promo_code_success = order.get_promo_code_success_message(False)
        promo_code_error = order.get_promo_code_error(False)
```

For now, you cannot clean the message from the session here, or existing databases where the server is restarted (without updating the module) won't show the messages at all.

You have to clean the session content manually, in the end of the route, after the rendering of the template.

Reply:
Updated, I'm not sure about the clear of the session If It's like this how you meant it. Atleast, It's working.

Reply:
Yes, that was the idea, but should be in `website_sale_loyalty` override to be in the module where this logic is defined :).

Reply:
Yes..you're right 😅 Thanks again for the review